File: restdocWithSwagger/src/main/java/com/example/restdocwithswagger/web/WebController.java
package com.example.restdocwithswagger.web;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.server.ServerErrorException;

import java.util.ArrayList;
import java.util.List;

@RestController
@RequestMapping("/web")
public class WebController {


    @RequestMapping("/list")
    public List<WebData> list(){

        List<WebData> list = new ArrayList<>();

        WebData webData = new WebData("deity", "name1");
        WebData webData2 = new WebData("deity2", "name2");

        list.add(webData);
        list.add(webData2);

        return list;
    }

    record WebData(String id, String name) {
    }

    @RequestMapping("/500")
    public void list500(){

        throw new ServerErrorException("서버오류가 발생하였습니다.");
    }

}

File: kotilWebSecurity/src/main/kotlin/kr/pe/deity/kotilwebsecurity/web/WebController.kt
package kr.pe.deity.kotilwebsecurity.web

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.server.ServerErrorException

@RestController
@RequestMapping("/web")
class WebController {

    //Request 확인

    @GetMapping("/get")
    fun get(): String = "Hello api"

    @GetMapping("/param")
    fun get(userId: String) = userId

    @GetMapping("/get/{userId}")
    fun getPathVariable(@PathVariable userId: String) = userId

    @PostMapping("/post/string")
    fun post(@RequestBody(required = false) userId: String) = userId

    @PostMapping("/post/dto")
    fun postDto(@RequestBody testRequest: TestRequest) = testRequest

    //Response 확인
    @GetMapping("/list")
    fun list(): ResponseEntity<DataResponse<List<WebResponse>>>{
        val webResponse = WebResponse("deity", "name")
        val webResponse2 = WebResponse("deity2", "name2")

        val list = listOf<WebResponse>(webResponse, webResponse2)

        return ResponseEntity.ok(DataResponse(data = list))

    }

    @GetMapping("/500")
    fun list500():Nothing = throw ServerErrorException("서버팀에 문의하세요")


}

data class WebResponse(val id:String, val name:String)

data class TestRequest(val id:String, val name:String)
